docs/design/gang_scheduling.md

```diff
@@ -0,0 +1,631 @@
+---
+id: gang_scheduling
+title: Gang scheduling design
+---
+
+<!--
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ -->
+# Gang Scheduling Implementation
+## Change log
+
+| Change Date | Editors               | Description                                    |
+| ----------- | --------------------- | ---------------------------------------------- |
+| 2020-11-25  | Wilfred Spiegelenburg | Initial document                               |
+| 2020-11-26  | Wilfred Spiegelenburg | Allocation release definition, order change    |
+| 2020-11-27  | Wilfred Spiegelenburg | Sequence diagram                               |
+| 2020-12-12  | Wilfred Spiegelenburg | Application submit and cleanup                 |
+| 2020-12-15  | Wilfred Spiegelenburg | AllocationRelease message and cleanup sequence |
+| 2020-12-24  | Wilfred Spiegelenburg | Allocation changes for recovery                |
+| 2021-02-15  | Wilfred Spiegelenburg | Placeholder allocation and ask cleanup         |
+
+YUNIKORN-2 describes a new format for scheduling applications by taking into account the overall demand the application will have.
+It guarantees the specified resources for the application by reserving the resources.
+
+There are two parts to this implementation:
+*   Kubernetes Shim
+*   Core and scheduling
+
+This document describes the implementation on the core side.
```

Comment:
having a design doc for the shim and another one for the core is quite confusing
I think we should put them all in this doc
After I read all the doc, since it explains a lot of the expected behaviors of the shim, I don't think we should exclude K8shim implementation out of this doc, suggest removing this part.

Reply:
Did you write a design for the shim after the major change to the design in November? The last doc version that I saw for the shim design was from before that time. In that version of the design the shim handled the placeholder allocation and the core was not aware of placeholders at all.
If we have a design that we implemented, or an updated design, that at least aligns with the implementation we can add it to this doc and update it.

Reply:
Agree, we can add that into this doc later.